### PR TITLE
[Draft] Style, Performance, and Compatibility Improvements

### PR DESCRIPTION
**Pull request by KPrasch**
Based over #15

### CODE REVIEW

File: monitor/dashboard.py [hunk 1]
```diff
@@ -64,68 +72,108 @@ def make_dash_app(monitor, flask_server: Flask, route_url: str, domain: str):
...
+            data = self.make_request()
+            data = data['activity']
+            confirmed, pending, inactive = data['active'], data['pending'], data['inactive']
+            total_stakers = confirmed + pending + inactive
+            return html.Div([html.H4("Active Ursulas"), html.H5(f"{confirmed}/{total_stakers}", id='active-ursulas-value')])
 
         @dash_app.callback(Output('staker-breakdown', 'children'), [Input('minute-interval', 'n_intervals')])
         def stakers_breakdown(n):
-            return stakers_breakdown_pie_chart(staking_agent=monitor.staking_agent)
+            data = self.make_request()
```
[the remark this answers is not in the document]
Comment (derekpierre) [reply monitor/dashboard.py]: ✔️ . Fixed via https://github.com/KPrasch/nucypher-monitor/pull/1

File: tests/cli/reports/pytest-results.xml [hunk 2]
```diff
@@ -0,0 +1,298 @@
+<?xml version="1.0" encoding="utf-8"?><testsuites><testsuite errors="0" failures="1" hostname="pegasus" name="pytest" skipped="0" tests="1" time="8.526" timestamp="2020-01-17T14:52:36.889923"><testcase classname="tests.cli.test_main" file="tests/cli/test_main.py" line="42" name="test_monitor_crawl_run" time="5.942"><failure message="web3.exceptions.BadFunctionCallOutput: Could not transact with/call contract function, is contract deployed correctly and chain synced?">cls = &lt;class &apos;nucypher.blockchain.eth.agents.ContractAgency&apos;&gt;
```
[the remark this answers is not in the document]
Comment (derekpierre) [reply tests/cli/reports/pytest-results.xml]: ✔️ . Fixed via https://github.com/KPrasch/nucypher-monitor/pull/1.

File: monitor/crawler.py [hunk 3]
```diff
@@ -53,30 +66,29 @@ def init_db_tables(self):
...
-        from nucypher.network.nodes import FleetStateTracker
-        state_dict = FleetStateTracker.abridged_state_details(state)
-        # convert updated timestamp format for supported sqlite3 sorting
-        state_dict['updated'] = state.updated.rfc3339()
-        db_row = (state_dict['nickname'], state_dict['symbol'], state_dict['color_hex'],
-                  state_dict['color_name'], state_dict['updated'])
+    def store_state_metadata(self, state: dict):
+        # TODO Limit the size of this table - no reason to store really old state values
+        # TODO: Convert updated timestamp format for supported sqlite3 sorting
+        # state['updated'] = state['updated'].rfc3339()
```
Comment (derekpierre) [inline monitor/crawler.py]: @KPrasch , this adjustment of `updated` format was how the previous states got properly sorted. SQLite allows for sorting based on the rfc3339 timestamp format.  See the query for previous states at https://github.com/nucypher/nucypher-monitor/blob/a7d2c3c6eaa4b4c3eb5efca9b209f4a03597db29/monitor/db.py#L47. The use of `ORDER BY datetime(updated) DESC LIMIT {limit}` was how the states were returned in order from newest to oldest and displayed right (newest) to left (oldest)

Comment (derekpierre) [reply monitor/crawler.py]: Alternatively, if not used, we'd have to sort the result of the query ourselves.

Comment (derekpierre) [reply monitor/crawler.py]: Addressed in `derekpierre/prev-states` branch

**Comment by derekpierre** [comment]
@KPrasch should we merge and then get some of our other work merged. 

Current outstanding work:
- https://github.com/KPrasch/nucypher-monitor/tree/treemap 
- https://github.com/derekpierre/nucypher-monitor/tree/past-future-graph (based on treemap)
- https://github.com/derekpierre/nucypher-monitor/tree/prev-states (based on past-future-graph)

Also since the nucypher `kprasch/monitoring` branch was merged in https://github.com/nucypher/nucypher/pull/1583, we should update our requirements to use `nucypher:master`

**Comment by KPrasch** [comment]
Moved to #26

**Pull request closed without merging**